base/applications/kbswitch/indicdll/indicdll.c

```diff
@@ -18,18 +21,18 @@ typedef struct tagSHARED_DATA
     HWND hKbSwitchWnd;
     UINT nHotID;
     DWORD_PTR dwHotMenuItemData;
-    CRITICAL_SECTION csLock;
```

Comment:
I wonder what the idea of the "shared" critical section lock was. To lock the shared data if there is _**more than one instance**_ of kbswitch/indicdll running (I mean, if it's possible to have multiple threads from multiple processes invoke the indicdll stuff and the shared data) ?
If this is _really_ the case, shouldn't a global named mutex of some sort (or unnamed but then its handle would have to be duplicated probably and passed around...??) be used instead?

Reply:
`indicdll.dll` is not only for `kbswitch.exe` (`internat.exe`).
`win32ss\user\user32\misc\imm.c` also uses `indicdll.dll`.